operator/src/main/java/oracle/kubernetes/operator/helpers/CrdHelper.java

```diff
@@ -330,6 +365,12 @@ Step updateExistingCrd(Step next, V1CustomResourceDefinition existingCrd) {
           existingCrd.getMetadata().getName(), existingCrd, createReplaceResponseStep(next));
     }
 
+    Step updateExistingCrdWithConversion(Step next, V1CustomResourceDefinition existingCrd) {
+      existingCrd.getSpec().conversion(createConversionWebhook());
+      return new CallBuilder().replaceCustomResourceDefinitionAsync(
```

Comment:
Not sure if I'm missing something here. The `onFailure` method  of the `ReplaceResponseStep` calls the onFailure of the parent with the `conflictStep` defined in the context. If the "two webhook instances" are part of the same deployment, then they'll share the same identity (certificate/key data). The identity data is generated only once and stored in the Webhook's configmap and secret. For subsequent calls, it reuses the same identity instead of creating a new identity.

Reply:
Once I've got the tests working on release/3.4 then I'd be interested to work on this problem :).  I think that you are missing that if two webhook replicas start concurrently then they may each try to create a certificate/key and then they will need to negotiate when one of them wins -- that is that the other will have to rewind and read the data shared by the other.

Reply:
ok, sure :) I think this is related to the discussion on slack thread. If two replicas are started concurrently, there'll be a race and winner will create certificate/key and write them to secret. However, the other instance (loser) might overwrite the certificate/key stored by the winner. I think we need some way to synchornize so that loser doesn't overwrite the certificate/key stored by winner.

Reply:
ok. I think this is likely resolved by 6b88d5b. I did some tests by setting 2 replicas for the webhook deployment and it seem to work fine.